xds/src/main/java/io/grpc/xds/client/ControlPlaneClient.java

```diff
@@ -194,7 +201,7 @@ void nackResponse(XdsResourceType<?> type, String nonce, String errorDetail) {
    */
   // Must be synchronized.
   boolean isInBackoff() {
-    return rpcRetryTimer != null && rpcRetryTimer.isPending();
+    return rpcRetryTimer != null || (hasBeenActive && !isReady());
```

Comment:
I have introduced an `isConnected()` for ControlPlaneClient.
I've changed this from `!isReady()` to `!xdsTransport.isConnected()`. 
There are 2 types of backoff:
  1. The retryTimer
  2. The subchannel reconnect backoff logic

We want to consider them the same, since in both cases we'll try to connect without any further explicit action.